internal/pkg/agent/cmd/run.go

```diff
@@ -148,6 +152,9 @@ func run(override cfgOverrider, testingMode bool, fleetInitTimeout time.Duration
 		"source": agentName,
 	})
 
+	// Make sure to flush any buffered logs before we're done.
+	defer l.Sync() //nolint:errcheck // flushing buffered logs is best effort.
```

Comment:
Wouldn't it be safer to do this on `baseLogger` in case that ends up being passed somewhere else?

Reply:
Done in 367dfbf41.